src/bin/mainnet-test.rs

```diff
@@ -279,7 +297,7 @@ async fn main() -> anyhow::Result<()> {
         thread::sleep(Duration::new(180, 0));
     }
 
-    if !args.persist_db {
+    if !persist_db {
```

Comment:
gotta remove database dir at the beginning 'stead of the end cuz of `ctrl-c`

Reply:
Beginning of what? You can't do this before the `for` loop and you shouldn't do it before the watch loop